lib/langchain/llm/openai.rb

```diff
@@ -69,7 +69,7 @@ def complete(prompt:, **params)
       return legacy_complete(prompt, parameters) if is_legacy_model?(parameters[:model])
 
       parameters[:messages] = compose_chat_messages(prompt: prompt)
-      parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model])
+      parameters[:max_tokens] = validate_max_tokens(parameters[:messages], parameters[:model], parameters[:max_tokens])
```

Comment:
```ruby
max_tokens = [max_tokens, options[:max_tokens]].min
```
Do you think this ^^ logic should actually live inside of this method and not inside the base_validator or the openai_validator?

For example we could have something like this:
```ruby
parameters[:max_tokens] = [
  parameters[:max_tokens],
  validate_max_tokens(parameters[:messages], parameters[:model])
].min
```

I guess what's missing with this approach is raising an error?

Reply:
@andreibondarev if `:max_tokens` is unique to OpenAI then definitely no need to bleed into the `BaseValidator`. Otherwise it will mean each LLM endpoint will need to implement similar code. What do you think?

Reply:
@andreibondarev also what error condition did you have in mind? e.g. if `:max_tokens` is passed in as a non-number value? Or possibly as a negative value?

Reply:
@andreibondarev check out the latest commit. I moved it into the `OpenAIValidator` subclass.

Reply:
> @andreibondarev also what error condition did you have in mind? e.g. if `:max_tokens` is passed in as a non-number value? Or possibly as a negative value?

This is the error that I was referring to: https://github.com/andreibondarev/langchainrb/pull/388/files#diff-79c9e673e647cc360e11c2946b3a4def6a5dfe52480860dc9ccd2101906902a9R30